armi/reactor/blueprints/tests/test_materialModifications.py

```diff
@@ -76,6 +84,93 @@ def test_u235_wt_frac_modification(self):
         u = fuelComponent.getMass("U")
         assert_allclose(0.20, u235 / u)
 
+        fuelComponent = a[0][1]
+        u235 = fuelComponent.getMass("U235")
+        u = fuelComponent.getMass("U")
+        assert_allclose(0.20, u235 / u)
+
+    def test_u235_wt_frac_blockwise_modification(self):
+        a = self.loadUZrAssembly(
+            """
+        material modifications:
+            blockwise:
+                U235_wt_frac: [0.20]
+        """
+        )
+        fuelComponent = a[0][0]
+        u235 = fuelComponent.getMass("U235")
+        u = fuelComponent.getMass("U")
+        assert_allclose(0.20, u235 / u)
+
+        fuelComponent = a[0][1]
+        u235 = fuelComponent.getMass("U235")
+        u = fuelComponent.getMass("U")
+        assert_allclose(0.20, u235 / u)
+
+    def test_u235_wt_frac_componentwise_modification1(self):
+        a = self.loadUZrAssembly(
+            """
+        material modifications:
+            componentwise:
+                fuel1:
+                    U235_wt_frac: [0.20]
+            U235_wt_frac: [0.30]
```

Comment:
Can you describe how this would work if both a general wt frac and a component wt frac are provided together? Does one invalidate the other or is there a check that we should have to fail or warn the user?

Also, maybe if we are going this route then does it even make sense to maintain block material modifications? Maybe from a backwards compatibility perspective, but I think that what we really want are just mods on the component level anyways. 

@ntouran / @john-science?

Reply:
Good question @jakehader . I added a section in the docs that describes how such a situation would work. Check [here](https://github.com/terrapower/armi/pull/517/files#diff-457ae5e548d65f136e8364a5191df93f944ca0b0734e8338f8dc0a6bbd2256fb).

The componentwise values supplant the blockwise values, if both are provided. Any value not provided componentwise will fall back to the blockwise value. And if neither are provided, then it will fall back to whatever the material class does (i.e. either use some default in the material class or fail).

Reply:
My opinion is that the blockwise values are still useful, just to provide that default behavior if a componentwise value isn't provided.